nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,1825 @@
+        "\n",
+        "**Disclaimer:** You need a key to make requests to OpenAI API. You can request a key [here](https://beta.openai.com/docs/developer-quickstart) or you just can load the synthetic samples that we generated in the next cell."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "BAzLn9t83YCS"
+      },
+      "source": [
+        "# define the number of synthetic samples to generate\n",
+        "n = 10\n",
+        "new_texts = []\n",
+        "new_labels = []\n",
+        "api_key =  # insert your api key for GPT-3\n",
+        "headers = {'Authorization' : 'Bearer ' + api_key,\n",
+        "              'Content-type':'application/json', \n",
+        "              'Accept':'application/json'}\n",
+        "\n",
+        "iter = 0\n",
+        "while iter < n:\n",
+        "    # select two random samples from training set\n",
+        "    text1, label1, text2, label2 = get_two_random_samples()\n",
+        "    # create the prompt\n",
+        "    prompt = get_prompt(text1, label1, text2, label2)\n",
+        "    # send a post request to gpt-3 using the prompt\n",
+        "    response = requests.post('https://api.openai.com/v1/engines/davinci/completions',\n",
+        "                             headers=headers,\n",
+        "                             data = json.dumps({\"prompt\": prompt, \n",
+        "                                                \"max_tokens\": 30,\n",
+        "                                                \"temperature\": 0.9,\n",
+        "                                                \"top_p\": 0.95}))\n",
+        "\n",
+        "    # get response and extract the generated text and label\n",
+        "    # the generated output will be in the form \"<text> (Sentiment: <label>)\"\n",
+        "    data = response.json()['choices'][0]['text'].split('\\n')[0].split('(Sentiment:')\n",
+        "\n",
+        "    if len(data) < 2:\n",
+        "        # the format of the response is invalid\n",
+        "        continue\n",
+        "\n",
+        "    text = data[0]\n",
+        "    label = data[1].split(')')[0].strip()\n",
+        "\n",
+        "    if label not in ['joy', 'anger', 'surprise']:\n",
+        "        # the format of the response is invalid\n",
+        "        continue\n",
+        "\n",
+        "    new_texts.append(text)\n",
+        "    new_labels.append(label2idx[label])\n",
+        "    iter += 1\n",
+        "\n",
+        "# define the synthetic dataset and save it to disk so as to prevent sending \n",
+        "# many api requests\n",
+        "synthetic_ds = Dataset.from_dict({'text': new_texts, 'label': new_labels})\n",
+        "synthetic_ds.save_to_disk('./drive/MyDrive/text_augmentation/gpt-3/10')"
```

Comment:
You seem to be saving them to a different location than you're loading them from in the next step, leading to this:
<img width="163" alt="image" src="https://user-images.githubusercontent.com/22496343/145835860-87db58f6-5f36-4cb6-80c1-046b47e51531.png">

So I would write them to './data/gpt-3/10' in this case

Reply:
parametrize the `10`, now it's hardcoded, which could result in writing results of the n=100 run being written to the wrong file